tests/test_plot.py

```diff
@@ -194,6 +200,30 @@ def test_pairplot_outputs():
     )
 
 
+def test_pairplot_with_gmm_inputs():
+    X = np.random.rand(15, 3)
+    gmm = GaussianMixture(n_components=3, covariance_type="full").fit(X)
+    labels = ["A"] * 5 + ["B"] * 5 + ["C"] * 5
+    # test data
+    with pytest.raises(TypeError):
+        pairplot_with_gmm(X="test", gmm=gmm)
+
+    with pytest.raises(ValueError):
+        pairplot_with_gmm(X=X, gmm=gmm, labels=["A"])
+
+    with pytest.raises(TypeError):
+        pairplot_with_gmm(X, gmm=None)
+
+
+def test_pairplot_with_gmm_outputs():
+    X = np.random.rand(15, 3)
+    gmm = GaussianMixture(n_components=3, covariance_type="full").fit(X)
+    labels = ["A"] * 5 + ["B"] * 5 + ["C"] * 5
+    fig = pairplot_with_gmm(X, gmm)
+    fig = pairplot_with_gmm(X, gmm, labels)
+    fig = pairplot_with_gmm(X, gmm, labels, title="Test")
```

Comment:
does matplotlib / seaborn have any testing functions somewhere in the library? I don't actually know how you'd write a proper test for plotting code

Reply:
https://youtu.be/LX2ksGYXJ80?t=5274

Reply:
@loftusa thank you! I will look into this!